### PR TITLE
QC check for balsamic analysis (fold-80 only)

### PR DESCRIPTION
**Pull request by keyvanelhami**
Please see issue #504 for instructions

### CODE REVIEW

File: BALSAMIC/utils/qc_check.py [hunk 1]
```diff
@@ -34,6 +34,34 @@ def read_qc_table(qc_table: dict):
...
+
+    Returns:
+        qc_df: List
+
+    """
+    fold80_value = list(hs_metrics)
+
+    return fold80_value
+
+def get_fold80(hs_metrics: pd.DataFrame) -> pd.DataFrame:
```
Comment (ashwini06) [inline BALSAMIC/utils/qc_check.py:51]:
```suggestion
```
duplicate function

**Review by ashwini06 — COMMENTED** [review]
remove duplicate function `get_fold80`

**Comment by hassanfa** [comment]
@keyvanelhami tests are failing with the following error:
```
    from BALSAMIC.utils.qc_check import read_hs_metrics, read_qc_table, check_qc_criteria, write_output
E   ImportError: cannot import name 'read_hs_metrics'
```

File: BALSAMIC/workflows/balsamic.smk [hunk 2]
```diff
@@ -372,14 +376,19 @@ if 'delivery' in config:
...
-        quality_control_results + analysis_specific_results
     output:
         os.path.join(get_result_dir(config), "analysis_finish")
     run:
-        import datetime
 
-        with open(str(output[0]), mode='w') as finish_file:
-            finish_file.write('%s\n' % datetime.datetime.now())
+        # Read the hs-metrics and get fold-80 value. Then check if the value pass/fail.
+        hs_metrics = read_json("/Users/keyvan.elhami/git_repositories/BALSAMIC/tests/test_data/qc_files/multiqc_picard_HsMetrics.json")
```
Comment (hassanfa) [inline BALSAMIC/workflows/balsamic.smk:384]: `"/Users/keyvan.elhami/git_repositories/BALSAMIC/tests/test_data/qc_files/multiqc_picard_HsMetrics.json"` is not correct. Use `input.hsmetrics[0]` instead. See Snakemake docuemtnation how to use input/output in a rule.

File: BALSAMIC/workflows/balsamic.smk [hunk 3]
```diff
@@ -372,14 +376,19 @@ if 'delivery' in config:
...
+        fold80_value = get_qc_value(hs_metrics)
+        check_result = check_qc_criteria_simple(fold80_value, "/Users/keyvan.elhami/git_repositories/BALSAMIC/tests/test_data/qc_files/fold_80.txt")
```
Comment (hassanfa) [inline BALSAMIC/workflows/balsamic.smk:386]: I don't think you can use `/Users/keyvan.elhami/git_repositories/BALSAMIC/tests/test_data/qc_files/fold_80.txt` here either. Pass the correct value instead. Use as much of `input`/`output`/`params` as you can.

File: BALSAMIC/workflows/balsamic.smk [hunk 4]
```diff
@@ -372,14 +376,19 @@ if 'delivery' in config:
...
+
+        # If QC failed raise an error
+        if not check_result:
+            raise Exception("Analysis failed")
+        else:
+            Path(output[0]).touch()
```
Comment (hassanfa) [inline BALSAMIC/workflows/balsamic.smk:392]: remove this and bring 

```
       with open(str(output[0]), mode='w') as finish_file:
            finish_file.write('%s\n' % datetime.datetime.now())
```

here

File: BALSAMIC/workflows/balsamic.smk [hunk 5]
```diff
@@ -372,14 +376,19 @@ if 'delivery' in config:
     FormatFile(delivery_ready)
 
 rule all:
-    input:
-        quality_control_results + analysis_specific_results
     output:
         os.path.join(get_result_dir(config), "analysis_finish")
     run:
-        import datetime
```
Comment (hassanfa) [inline BALSAMIC/workflows/balsamic.smk:380]: Keep this

**Review by hassanfa — CHANGES_REQUESTED** [review]
👌🏽  needs some ironing out, but getting closer.

Please also add some tests for the new functions.

**Comment by ivadym** [comment]
It will be addressed in #707.

**Pull request closed without merging**